src/nginx/module.cc

```diff
@@ -133,12 +133,10 @@ void ngx_esp_override_backend_path(ngx_http_request_t *r,
                               " " + ngx_str_to_std(r->http_protocol),
                           &r->request_line);
 
-    std::size_t found = backend_path.find_first_of('?');
-    if (found != std::string::npos) {
-      std::string uri;
-      url_decode(backend_path.substr(0, backend_path.find_first_of('?')), uri);
-      ngx_str_copy_from_std(r->pool, uri, &r->uri);
-    }
+    std::string decoded_uri;
```

Comment:
There is no test case without "?", right?

what is the main purpose of this change?

Reply:
No test case.  By reading this code,  I fell if backend_path() is changed,  and it doesn't have ? query parameter,  r->uri field is NOT modified.  

It may or may not important.  But I feel it should be updated to be the same as r->unpraseduri field.